### ISSUE TITLE
Clist_blind.dll: Use SysTreeView32 instead of ListBox

### ISSUE THREAD

**Issue by winman3000** [miranda-ng/miranda-ng#2322]
Currently, the ListBox class is used to display the contact list. To make it even easier to use with screen readers, the class SysTreeView32 is recommended. This also makes navigation through the contacts much easier. Furthermore, it is much more ergonomic for blind people to see whether a group is collapsed or expanded. Because it is a standard Windows class, the screen reader already knows this class.

**Comment by georgehazan** [comment]
perhaps you are right, but I would greatly appreciate the opinion of another users of clist_blind.

**Comment by georgehazan** [comment]
Michal Badin answered:
>i checked now my contact list. my screen reader says, the following. "group: facebook (expanded", i press down arrow, and see the contacts, with theyr online statuses. if i press left arrow on the groups name, it says "collapsed". so obviously, i have to expand it, to see the names. as for myself, i dont have problems using as it is now. i dont see, the problem, or point, why the poster wants this.

**Comment by winman3000** [comment]
@georgehazan: I'm a Braille user. The problem is not that JAWS does not announce something correctly, the problem is that it displays it incorrectly on the Braille display. With the proposed control, the display would be more correct on the Braille display without making adjustments to the screen reader. The display for "Collapsed" would be a + sign on the Braille display and for "Expanded" a - sign. Currently, everything is written out on the Braille display and takes up a lot of space.

It would also make navigation using letters easier. With the new, modified contact list, it is now the case that as soon as I press a letter to find a contact faster, all groups are expanded, and I don't want that.

**Comment by winman3000** [comment]
If my suggestion is accepted, I'm willing to test the "new" contact list before it is added to Miranda NG. Not that anything will be broken because of me.

**Comment by georgehazan** [comment]
yes, because in fact Miranda has thousands people who use it, I was greatly surprised by this fact

**Comment by winman3000** [comment]
If this proposal is too risky, I suggest to develop an alternative clist_blind which contains my proposal. This could be developed under clist_blind_new or clist_blind_alternative.

Should this not work either, I am willing to do without it, but then we would have to adjust the behaviour of the list a bit.

I do not want to make anything more complicated than it is, it was just a suggestion of mine.

**Comment by Piciok** [comment]
@georgehazan This reminds me of discussions we had back in 2013 where we were wondering what to do in order to make it possible for contacts to bew selectable checkboxes. This would be useful E.G. to select multiple contacts to create a group chat, mark the ones you would like to be ignored etc. Currently there is no way to accomplish this with ClistBlind. I recall someone proposing back then that perhaps replacing the list view with a tree view could solve the problem. Meanwhile, I really enjoy the ability to filter the contact list by typing a couple letters of the name and I would not like for it to go away, so perhaps turning this behaviour into a toggle would work?
I also understand @winman3000's argument. The more properties are conveyed through native controls and their states, the more power is given to the screen reader and its user to present this information in a custom, convenient way. Currently, as list boxes do not support expanding and collapsing things natively, the information has to be explicitly written to be accessible and there is not much that can be customized about it.

@winman3000: I know it's a dirty workaround but what you could do is modify the template of groups so that the info you need, such as state, is reported at the beginning and then either create a Miranda-specific dictionary in your screen reader or modify your langpack to show the state in an abbreviated, Braille-friendly form.

**Comment by winman3000** [comment]
@Piciok:

I'm afraid you have misunderstood me a little. I love the possibility to navigate by typing letters and I wouldn't want to miss it. I just don't like the fact that closed groups are expanded in the process. If a group is closed, I don't want to make the contact that is in the group visible with the letter input. However, this behavior could be made optional, as you suggested.

I really like the idea with the control boxes. Alternatively, you could also use the CTRL+Space key to select and deselect the contacts, as in Windows. The screen reader would then only have to be informed about the status of the control, whether an item has been selected or deselected.
